phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataRegionObserver.java

```diff
@@ -197,10 +203,15 @@ public void run() {
         // inProgress is to prevent timer from invoking a new task while previous one is still
         // running
         private final static AtomicInteger inProgress = new AtomicInteger(0);
+        private static final String UPDATE_DISABLED_TS = "UPSERT INTO " + SYSTEM_CATALOG + "(" + TENANT_ID + ","
+                + TABLE_SCHEM + "," + TABLE_NAME + "," + INDEX_DISABLE_TIMESTAMP + ") values (?,?,?,?)";
         RegionCoprocessorEnvironment env;
+        private long rebuildIndexBatchSize = HConstants.LATEST_TIMESTAMP;
 
         public BuildIndexScheduleTask(RegionCoprocessorEnvironment env) {
             this.env = env;
+            this.rebuildIndexBatchSize = env.getConfiguration().getLong(
```

Comment:
The configuration is confusing it's similar to number of mutations but it's actually a timestamp. It can be something like INDEX_FAILURE_HANDLING_REBUILD_PERIOD?